Makefile

```diff
@@ -20,7 +20,7 @@
 IMAGE_NAME = pulsar-client-go-test:latest
 PULSAR_VERSION ?= 2.10.3
 PULSAR_IMAGE = apachepulsar/pulsar:$(PULSAR_VERSION)
-GO_VERSION ?= 1.18
+GO_VERSION ?= 1.20
```

Comment:
@RobertIndie If I upgrade it in another PR, this PR will fail the CI.

Reply:
This will break the CI and will break the compatibility for the older go version. Could you implement it another way? Or do you think there is a strong motivation that we need to upgrade the go version?

Reply:
I didn't mean to upgrade the go version,  it just so happends that I use go 1.20 at the same time, I implement this is PR just in the purpose of retrying and when I test my retrying demo and found that I can not retry all the failed requests, some errors should fail fast instead of retrying(.e.g ErrSendQueueIsFull).

And I just found #1142 today, and found that this is a common requirement. I've checked the 3rd party packages, "https://github.com/cockroachdb/errors" require go 1.19, "https://github.com/hashicorp/go-multierror" has a different package name `multierror`, "https://github.com/uber-go/multierr" require go 1.20.

For the sake of compatibility, may be the simplest way is to implement errors.Join() in pulsar-client-go, what I mean is to copy the code in go SDK into our package, I will try that tomorrow.